pages/curriculum/[lesson].tsx

```diff
@@ -15,12 +16,20 @@ import _ from 'lodash'
 
 const Challenges: React.FC<QueryDataProps<AppData>> = ({ queryData }) => {
   const { lessons, session, alerts } = queryData
-  const userSubmissions: UserSubmission[] = _.get(session, 'submissions', [])
-  const lessonStatus: LessonStatus[] = _.get(session, 'lessonStatus', [])
   const router = useRouter()
   const currentlessonId = router.query.lesson as string
+  if (!session || !lessons || !alerts) {
+    return (
+      <Error title="Internal server error" message="Bad data" src="/500.png" />
+    )
+  }
+  if (!lessons.find((lesson: Lesson) => lesson.id === currentlessonId)) {
```

Comment:
`lessons.find((lesson: Lesson) => lesson.id === currentlessonId)` 

This line is repeated on line 32. You can clean this up by declaring `currentLesson` before this `if` statement.
```
const currentLesson: Lesson | undefined = lessons.find(
    (lesson: Lesson) => lesson.id === currentlessonId	
)
if (!currentLesson) {
         ........
}
```

Reply:
I removed duplicate code in both cases.